src/compiler/checker.ts

```diff
@@ -16217,9 +16216,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return typeParameter.constraint === noConstraintType ? undefined : typeParameter.constraint;
     }
 
-    function getParentSymbolOfTypeParameter(typeParameter: TypeParameter): Symbol | undefined {
+    function getEffectiveTypeParameterHost(typeParameter: TypeParameter) {
         const tp = getDeclarationOfKind<TypeParameterDeclaration>(typeParameter.symbol, SyntaxKind.TypeParameter)!;
-        const host = isJSDocTemplateTag(tp.parent) ? getEffectiveContainerForJSDocTemplateTag(tp.parent) : tp.parent;
+        return isJSDocTemplateTag(tp.parent) ? getEffectiveJSDocHost(tp.parent) : tp.parent;

```

Comment:
This sort of makes me wonder when we really should be using `getEffectiveJSDocHost` where we are currently using `getEffectiveContainerForJSDocTemplateTag`...

Reply:
Right, but to be fair - I had to use `getEffectiveJSDocHost` to fix this issue at hand and it didn't look to me like this older function would have to stick to the other one. There is a chance that it should not use this - but there is also a chance that this change fixed some other issue ;p 

In the ideal world, I'd go through all of the callers to both and assess their needs but I just don't have time to do that right now 😢